src/unix/fs.c

```diff
@@ -844,15 +846,9 @@ static ssize_t uv__fs_buf_iter(uv_fs_t* req, uv__fs_buf_iter_processor process)
     total += result;
   }
 
-  if (errno == EINTR && total == -1)
-    return total;
-
   if (bufs != req->bufsml)
     uv__free(bufs);
 
-  req->bufs = NULL;
-  req->nbufs = 0;
-
```

Comment:
Don't remove this. We don't want to leave a dangling pointer in the request structure.